lib/GenerateResource.js

```diff
@@ -0,0 +1,171 @@
+
+    if (options && options.xliffsDir) {
+        this.xliffsDir = options.xliffsDir;
+    }
+    if (options &&  options.resRoot) {
+        this.resRoot = options.resRoot;
+    }
+    if (options && options.resName) {
+        this.resName = options.resName;
+    }
+    this.ts = new TranslationSet(this.sourceLocale);
+};
+
+var xliffFileFilter = /([.*][\-])?([a-z][a-z][a-z]?)(-([A-Z][a-z][a-z][a-z]))?(-([A-Z][A-Z]))?\.xliff$/;
+
+/**
+ * Initialize this repository and read in all of the strings.
+ *
+ * @param {Project} project the current project
+ * @param {Function(Object, Object)} cb callback to call when the
+ * initialization is done
+ */
+GenerateResource.prototype.init = function() {
+    var dir = this.xliffsDir;
+    var list;
+    try {
+        list = fs.readdirSync(dir);
+    } catch (err) {
+        logger.warn(dir + "is invalid directory");
+    }
+    
+    if (list) {
+        list.filter(function(file) {
+            var match = xliffFileFilter.exec(file);
+            if (!match ||
+               match.length < 2 ||
+               (match.length >= 3 && match[2] && !utils.iso639[match[2]]) ||
+               (match.length >= 5 && match[4] && !utils.iso15924[match[4]]) ||
+               (match.length >= 7 && match[6] && !utils.iso3166[match[6]])) {
+                return false;
+            }
+            return true;
+        }).forEach(function (file) {
+            pathName = path.join(dir, file);
+            var xliff = new Xliff({
+                sourceLocale: this.sourceLocale
+            });
+            if (fs.existsSync(pathName)) {
+                var data = fs.readFileSync(pathName, "utf-8");
+                xliff.deserialize(data);
+                var ts = xliff.getTranslationSet();
+                this.ts.addSet(ts);
+            } else {
+                logger.warn("Could not open xliff file: " + pathName);
+            }
+        }.bind(this));
+    }
+};
+
+GenerateResource.prototype._calResourcePath = function(resLocale) {
+    var resPath = "";
+    var splitLocale = resLocale.split("-");
+    var minimalLocale = new LocaleMatcher({locale: resLocale}).getLikelyLocaleMinimal().getSpec();
+    if (resLocale === minimalLocale ) {
+        resPath = splitLocale[0];
+    } else {
+        resPath = splitLocale.join("/");
+    }
+    return resPath;
+}
+
+GenerateResource.prototype.getResourceFilePath = function(resLocale) {
+    var resPath = this._calResourcePath(resLocale);
+    return path.join(this.targetDir, this.resRoot, resPath, this.resName);
+}
+
+GenerateResource.prototype._writeFiles = function() {
+    var fullPath;
+    for (locale in this.output){
+        var resourcePath = this._calResourcePath(locale);
+        var content = JSON.stringify(this.output[locale], true, 4);
+
+        utils.makeDirs(path.join(this.targetDir, this.resRoot, resourcePath));
+        var fullPath = this.getResourceFilePath(locale, resourcePath);
+
+        fs.writeFileSync(fullPath, content, "utf-8");
```

Comment:
This still only outputs json files. I thought you had reworked this to use the plugins? Maybe you didn't commit or push those changes yet?

Reply:
Probably we had a different thought. I wanted to remove the part that creates the `ilibmanifest.json` file. It executes in a `close` task.  I have a newly defined localization cycle. https://github.com/iLib-js/loctool/blob/generateMode/loctool.js#L534
but Within there, `extract` and `write` process are skipped in generate mode. so that's why the code that creates the files in generateResource class is included..